### PR TITLE
build: android: build rust code with gradle

### PR DESCRIPTION
**Pull request by montekki**
Attempt to move `build.sh` logic to gradle scripts.

### CODE REVIEW

**Comment by montekki** [comment]
for `build.rs` to work `Cargo.toml` flies from `*-hack` directories obviously have to contain

```toml 
[build-dependencies]
definitions = {path = "../definitions"}
generate_message = {path = "../generate_message"}

```

File: android/app/build.gradle [hunk 1]
```diff
@@ -62,18 +88,18 @@ dependencies {
...
+	implementation 'androidx.compose.runtime:runtime-livedata:1.1.1'
     implementation 'androidx.security:security-crypto:1.1.0-alpha03'
     implementation 'androidx.biometric:biometric:1.1.0'
 	implementation 'androidx.security:security-crypto-ktx:1.1.0-alpha03'
     implementation 'androidx.camera:camera-core:1.0.2'
     implementation 'androidx.camera:camera-lifecycle:1.0.2'
-	implementation 'androidx.camera:camera-view:1.1.0-beta01'
-	implementation 'androidx.camera:camera-camera2:1.1.0-beta01'
+	implementation 'androidx.camera:camera-view:1.1.0-beta02'
+	implementation 'androidx.camera:camera-camera2:1.1.0-beta02'
```
Comment (Slesarew) [inline android/app/build.gradle:102]: TODO: test on several devices, check changelog. This is known to break a bit occasionally.

Comment (montekki) [reply android/app/build.gradle]: actually reverted the bump, let's make this changeset minimal

**Comment by montekki** [comment]
goto #975

**Pull request closed without merging**